src/radiation_system.hpp

```diff
@@ -917,189 +1022,244 @@ void RadSystem<problem_t>::AddSourceTerms(array_t &consVar, arrayconst_t &radEne
 		const double Egastot0 = consPrev(i, j, k, gasEnergy_index);
 		auto massScalars = RadSystem<problem_t>::ComputeMassScalars(consPrev, i, j, k);
 
+		// ------  CCH: Multigroup ---------
 		// load radiation energy
-		const double Erad0 = consPrev(i, j, k, radEnergy_index);
+		quokka::valarray<double, nGroups_> Erad0Vec;
+		for (int g = 0; g < nGroups_; ++g) {
+			Erad0Vec[g] = consPrev(i, j, k, radEnergy_index + numRadVars_ * g);
+		}
+		AMREX_ASSERT(min(Erad0Vec) >= 0.0);
+		const double Erad0 = sum(Erad0Vec);
 
 		// load radiation energy source term
 		// plus advection source term (for well-balanced/SDC integrators)
-		const double Src = dt * ((chat * radEnergySource(i, j, k)) + advectionFluxes(i, j, k));
+		quokka::valarray<double, nGroups_> Src;
+		for (int g = 0; g < nGroups_; ++g) {
+			Src[g] = dt * (chat * radEnergySource(i, j, k, g) + advectionFluxes(i, j, k, g));
+		}
 
 		double Egas0 = NAN;
 		double Ekin0 = NAN;
 		double Egas_guess = NAN;
-		double Erad_guess = NAN;
 		double T_gas = NAN;
+		quokka::valarray<double, nGroups_> EradVec_guess{};
+		quokka::valarray<double, nGroups_> kappaPVec{};
+		quokka::valarray<double, nGroups_> kappaEVec{};
+		quokka::valarray<double, nGroups_> kappaFVec{};
 
-		if constexpr (gamma_ != 1.0) {
+		for (int g = 0; g < nGroups_; ++g) {
+			EradVec_guess[g] = NAN;
+		}
+
+		// if constexpr (gamma_ != 1.0) {
+		if (gamma_ != 1.0) {
```

Comment:
Does this not work with `constexpr` anymore?

Reply:
`constexpr` works now. I've added it back.